virtual_labs/tests/projects/conftest.py

```diff
@@ -139,3 +139,30 @@ async def mock_create_vl_projects(
     for elt in vl_projects:
         for vl_id, projects in elt.items():
             await cleanup_resources(client, vl_id)
+
+
+@pytest_asyncio.fixture
+async def mock_create_full_vl_projects(
+    async_test_client: AsyncClient,
+) -> AsyncGenerator[tuple[list[dict[str, Any]], dict[str, str]], None]:
+    client = async_test_client
+    headers = get_headers()
+
+    virtual_lab_resp = await create_mock_lab(async_test_client)
+    virtual_lab_id = virtual_lab_resp.json()["data"]["virtual_lab"]["id"]
+    projects: list[dict[str, Any]] = []
```

Comment:
Not a big deal but type safety could be further improved by using a TypedDict: https://docs.python.org/3/library/typing.html#typing.TypedDict


Up to you to decide if we do this now or add a TODO and do it later.

Reply:
good one, I will make it as todo, since all our test are not fully-typed, so this change must also be applied to all the test